vite.config.ts

```diff
@@ -14,6 +16,37 @@ export default defineConfig({
   },
   build: {
     target: 'esnext',
+    rollupOptions: {
+      input: {
+        main: 'index.html',
+        'service-worker': './src/service-worker.js',
+      },
+      output: {
+        entryFileNames: (chunkInfo) => {
+          return chunkInfo.name === 'service-worker'
+            ? '[name].js'
+            : 'assets/[name]-[hash].js'
+        },
+      },
+      plugins: [
+        {
+          name: 'copy-pwa-assets',
+          generateBundle() {
+            const pwaAssetsDir = path.resolve(__dirname, 'pwa-assets')
+            if (fs.existsSync(pwaAssetsDir)) {
+              const files = fs.readdirSync(pwaAssetsDir)
+              for (const file of files) {
+                this.emitFile({
+                  type: 'asset',
+                  fileName: `pwa-assets/${file}`,
+                  source: fs.readFileSync(path.resolve(pwaAssetsDir, file)),
+                })
```

Comment:
pwa-asset-generator doesn't allow us to specify a reference path for assets separate from where they're actually placed (ie: assets placed in dist are referenced as /dist/pwa-assets by pwa-asset-generator's auto-generated code)